### DISCUSSION TITLE
Importing existing postgress database into Vercel storage

### DISCUSSION THREAD

**Discussion by Siarune** [vercel/community#2257] [Help]
### Summary

I'm currently trying to use pg_dump and pg_restore to transfer my database from Supabase to Vercel's postgres. 
The dump appears to work fine, but the restore hangs.

```
pg_dump --data-only --format=tar --schema=public '<db connection string>' > postgres.tar
pg_restore --host=<db connection string> --file=postgres.tar --data-only
```

### Additional information

No errors thrown, but the Vercel dashboard showed a substantial amount of compute time (1.2 hrs) that indicates it at least had a connection.

**Comment by sebastianroming** [comment]
Hi @Siarune,

Importing data from your existing PostgreSQL database to Vercel Postgres should be possible using only the `pg_dump` and `psql` command - without the need of `pg_restore`, e.g. `pg_dump <connection-string> | psql <connection-string>`.

```
pg_dump postgres://<user>:<pass>@<hostname>:5432/mydb | psql postgres://default:<secret>@<vercel_storage_host>:5432/verceldb
```

**Reply by Siarune** [reply]
The reason I used `pg_dump` in the first place is because of a foreign key constraint, In which I used the `--disable-triggers` flag. As far as I know, the `psql` command doesn't have an equivalent.

**Reply by sebastianroming** [reply]
Hi @Siarune!
Did you run the commands from your local machine or the integrated CLI in your Vercel Dashboard?
`pg_dump` is necessary, that's correct. But you should be good to run the import using `psql` instead of `pg_restore` – can you give it a try, without exporting it to a tar file?

**Reply by Siarune** [reply]
This was run from a local machine,  `pg_dump <connection_string> --dataonly | psql <connection_string>`, which returns:

```
pg_dump: warning: there are circular foreign-key constraints on this table:
SET
SET
SET
SET
SET
 set_config 
------------
 
(1 row)

SET
SET
SET
SET
COPY 0
COPY 0
COPY 0
COPY 0
COPY 0
COPY 0
COPY 0
COPY 0
COPY 0
COPY 0
COPY 0
COPY 0
COPY 0
ERROR:  duplicate key value violates unique constraint "schema_migrations_pkey"
DETAIL:  Key (version)=(20171026211738) already exists.
CONTEXT:  COPY schema_migrations, line 1
COPY 0
ERROR:  relation "pgsodium.key" does not exist
invalid command \N
invalid command \.
ERROR:  syntax error at or near "ebcccd4a"
LINE 1: ebcccd4a-e77d-4bc0-b684-b248a90b1b67 default 2022-10-11 19:5...
        ^
invalid command \n100
invalid command \nwon't
invalid command \nwent
invalid command \n
invalid command \n[Urmom](https://urmom6969.top)
invalid command \nand
invalid command \nI
invalid command \nHe
invalid command \
invalid command \nare
invalid command \n
invalid command \nwhen
invalid command \n
invalid command \ndoes
invalid command \nI
invalid command \n
invalid command \n
invalid command \nIn
invalid command \n
invalid command \nThe
invalid command \.
ERROR:  syntax error at or near "11"
LINE 1: 11 2022-10-29 22:46:58.918 2022-12-02 22:53:30.027 100 bugs ...
        ^
invalid command \.
ERROR:  syntax error at or near "1"
LINE 1: 1 2022-10-28 23:05:46.326 2023-04-09 02:42:03.981 Aidan aida...
        ^
invalid command \.
ERROR:  syntax error at or near "6"
LINE 1: 6 2023-04-17 21:26:49.087 2023-04-17 21:33:13.882 1 54 hi Ai...
        ^
invalid command \.
ERROR:  syntax error at or near "6"
LINE 1: 6 2022-12-02 23:00:12.772 2022-12-02 23:00:12.772 2023-01-01...
        ^
invalid command \.
COPY 0
ERROR:  duplicate key value violates unique constraint "migrations_name_key"
DETAIL:  Key (name)=(create-migrations-table) already exists.
CONTEXT:  COPY migrations, line 1
COPY 0
 setval 
--------
      1
(1 row)

 setval 
--------
      7
(1 row)

 setval 
--------
     57
(1 row)

 setval 
--------
     42
(1 row)

 setval 
--------
      1
(1 row)

 setval 
--------
      4
(1 row)
```

The errors seem to be coming from:
- "Invalid" markdown strings in cells `\`
   Conflicts with psql's command scheme that starts with `\`
- Duplicate key value in the prisma migration table, which is deleted in the original database
- Absence of `pgsodium.key`
- Syntax error in "line 1" `ebcccd4a-e77d-4bc0-b684-b248a90b1b67 default 2022-10-11 19:5...`

**Comment by sun-ks** [comment]
I am going to export my Postgres to set up it on my local pc.  Is it possible?
